stmhal/storage.c

```diff
@@ -56,6 +56,13 @@ STATIC byte flash_cache_mem[0x4000] __attribute__((aligned(4))); // 16k
 #define FLASH_MEM_SEG1_START_ADDR (0x08004000) // sector 1
 #define FLASH_MEM_SEG1_NUM_BLOCKS (128) // sectors 1,2,3,4: 16k+16k+16k+16k(of 64k)=64k
 
+#elif defined(STM32F429xx)
+
+#define CACHE_MEM_START_ADDR (0x10000000) // CCM data RAM, 64k
+#define FLASH_SECTOR_SIZE_MAX (0x10000) // 64k max, size of CCM
+#define FLASH_MEM_SEG1_START_ADDR (0x08004000) // sector 1
+#define FLASH_MEM_SEG1_NUM_BLOCKS (224) // sectors 1,2,3,4: 16k+16k+16k+64k=112k
```

Comment:
If you don't mind wasting half the flash, you can start this at sector 12 and make it use the last 1mb of flash (which will be 512k for the fatfs due to the 64k cache size).  You could also use the SDRAM for the cache and then utilise all the 1mb of flash for the fatfs.